streams/src/test/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImplTest.java

```diff
@@ -177,4 +182,132 @@ public void shouldReduceWithInternalStoreName() {
+
+        processData(topic);
+        final KeyValueStore<String, String> aggregate = (KeyValueStore<String, String>) driver.allStateStores().get("aggregate");
+        assertThat(aggregate.get("1"), equalTo("0+1+1+1"));
+        assertThat(aggregate.get("2"), equalTo("0+2+2"));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointOnCountWhenMaterializedIsNull() {
+        groupedTable.count((Materialized) null);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerOnReduceWhenMaterializedIsNull() {
+        groupedTable.reduce(MockReducer.STRING_ADDER, MockReducer.STRING_REMOVER, (Materialized) null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerOnReduceWhenAdderIsNull() {
+        groupedTable.reduce(null, MockReducer.STRING_REMOVER, Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("store"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerOnReduceWhenSubtractorIsNull() {
+        groupedTable.reduce(MockReducer.STRING_ADDER, null, Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("store"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerOnAggregateWhenInitializerIsNull() {
+        groupedTable.aggregate(null,
+                               MockAggregator.TOSTRING_ADDER,
+                               MockAggregator.TOSTRING_REMOVER,
+                               Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("store"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerOnAggregateWhenAdderIsNull() {
+        groupedTable.aggregate(MockInitializer.STRING_INIT,
+                               null,
+                               MockAggregator.TOSTRING_REMOVER,
+                               Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("store"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerOnAggregateWhenSubtractorIsNull() {
+        groupedTable.aggregate(MockInitializer.STRING_INIT,
+                               MockAggregator.TOSTRING_ADDER,
+                               null,
+                               Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("store"));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerOnAggregateWhenMaterializedIsNull() {
+        groupedTable.aggregate(MockInitializer.STRING_INIT,
+                               MockAggregator.TOSTRING_ADDER,
+                               MockAggregator.TOSTRING_REMOVER,
+                               (Materialized) null);
+    }
+
+    private void processData(final String topic) {
```

Comment:
`doShouldReduce` could call this function to reduce duplicated code?

Reply:
Not without changing the tests that use `doShouldReduce` - perhaps worth doing separately